### PR TITLE
Avoid use of Delayed in to_parquet

### PR DESCRIPTION
**Pull request by mrocklin**
This removes our ability to perform graph optimization, which can negatively impact memory use in complex scenarios.

Instead we use a dask.dataframe.Scalar class here.  We should probably do this elsewhere as well.

FYI @martindurant @rjzamora @ian-r-rose

### CODE REVIEW

**Review by rjzamora — APPROVED** [review]
Makes sense to me!

File: dask/dataframe/io/parquet/core.py [hunk 1]
```diff
@@ -710,10 +711,10 @@ def to_parquet(
             {"append": append, "compression": compression},
         )
     else:
-        dsk[name] = (lambda x: None, part_tasks)
+        dsk[(final_name, 0)] = (lambda x: None, part_tasks)
 
     graph = HighLevelGraph.from_collections(name, dsk, dependencies=[df])
-    out = Delayed(name, graph)
+    out = Scalar(graph, final_name, "")
```
Comment (rjzamora) [inline dask/dataframe/io/parquet/core.py:717]: Looks like the docstring for to_parquet still says the return will be `Delayed` if `compute=False`

**Comment by mrocklin** [comment]
(Thanks @jrbourbeau )

**Review by jrbourbeau — COMMENTED** [review]
Thanks for the improvement @mrocklin and reviewing @rjzamora. The test failures we're seeing here are due to deprecations in the recent `pyarrow=5.0.0` release. I've opened https://github.com/dask/dask/pull/7960 to temporarily pin `pyarrow<5` in CI. Once that's in, I'll merge `main` here to unblock CI

**Pull request merged**